osu.Game/Screens/OnlinePlay/Components/RoomManager.cs

```diff
@@ -98,7 +98,9 @@ public virtual void PartRoom()
             if (JoinedRoom.Value == null)
                 return;
 
-            api.Queue(new PartRoomRequest(joinedRoom.Value));
+            if (api.State.Value == APIState.Online)
```

Comment:
Without this check is it actually feasible that this fires and causes issues?

Reply:
Yes, I hit this once in testing.

Reply:
Looking at this again, I think there should be no issues with this. Server-side flows are supposed to correctly clean up the room anyway, so this request not being realised should have no impact on the server.